src/react/databrowser/objects/ObjectLockSetting.jsx

```diff
@@ -0,0 +1,311 @@
+//@flow
+import React, { useEffect } from 'react';
+import { useParams } from 'react-router';
+import { useDispatch, useSelector } from 'react-redux';
+import { FormProvider, useForm, Controller } from 'react-hook-form';
+import { push } from 'connected-react-router';
+import { DateTime } from 'luxon';
+import * as JoiImport from '@hapi/joi';
+import DateExtension from '@hapi/joi-date';
+import { joiResolver } from '@hookform/resolvers';
+import { Button } from '@scality/core-ui/dist/next';
+import { Banner, Toggle } from '@scality/core-ui';
+import { SmallerText } from '@scality/core-ui/dist/components/text/Text.component';
+import SpacedBox from '@scality/core-ui/dist/components/spacedbox/SpacedBox';
+import FormContainer, * as F from '../../ui-elements/FormLayout';
+import {
+  clearError,
+  getObjectMetadata,
+  putObjectRetention,
+} from '../../actions';
+import { useQuery } from '../../utils/hooks';
+import {
+  getDefaultMinRetainUntilDate,
+  getDefaultRetention,
+  getRetainUntilDateHint,
+} from './utils';
+import type { AppState } from '../../../types/state';
+
+const Joi = JoiImport.extend(DateExtension);
+
+const objectLockRetentionSettingsValidationRules = {
+  isRetentionEnabled: Joi.boolean().default(false),
+  retentionMode: Joi.when('isRetentionEnabled', {
+    is: Joi.equal(true),
+    then: Joi.string().required(),
+    otherwise: Joi.valid(),
+  }),
+  retentionUntilDate: Joi.date()
+    .format('YYYY-MM-DD')
+    .min('now')
+    .raw(),
+};
+
+const schema = Joi.object(objectLockRetentionSettingsValidationRules);
+
+// Since IAM role permission hasn't been implemented, we only provide the minimum ability to edit the retention from the UI
+// and limit the following actions:
+// 1. When the retention is disabled by default, it's not allowed to enable it.
+// 2. The default retention mode is COMPLIANCE, it's not allowed to switch to the GOVERNANCE mode.
+export default function ObjectLockSetting() {
+  const dispatch = useDispatch();
+  const query = useQuery();
+  const hasError = useSelector(
+    (state: AppState) =>
+      !!state.uiErrors.errorMsg && state.uiErrors.errorType === 'byComponent',
+  );
+  const errorMessage = useSelector(
+    (state: AppState) => state.uiErrors.errorMsg,
+  );
+  const objectMetadata = useSelector(
+    (state: AppState) => state.s3.objectMetadata,
+  );
+  const { bucketName: bucketNameParam } = useParams();
+  const objectKey = query.get('prefix');
+  const versionId = query.get('versionId');
+  const {
+    isDefaultRetentionEnabled,
+    defaultRetentionMode,
+    defaultRetentionUntilDate,
+  } = getDefaultRetention(objectMetadata);
+
+  useEffect(() => {
+    dispatch(getObjectMetadata(bucketNameParam, objectKey, versionId));
+  }, [dispatch, bucketNameParam, objectKey, versionId]);
+
+  const useFormMethods = useForm({
+    mode: 'all',
+    resolver: joiResolver(schema),
```

Comment:
I think that to address my above comment you can likely passin the minimum date here